include/Common.h

```diff
@@ -16,6 +19,25 @@ namespace Common {
 
     int RandomInt(int min, int max);
 
+    template <typename T>
+    T RandomEvent(std::vector<int> weights, std::vector<T> outcomes) {
+        std::vector <int> events;
+        int k = 0;
+        for (auto it : weights)
```

Comment:
* use `int` instead of `auto`
* rename `it` to a better name, like `weight`
* use this for style: 
  ```
  for () {
  }
  ```